aodndata/acorn/handler.py

```diff
@@ -0,0 +1,133 @@
+
+        elif product_type == 'wavep' and file_version == 'FV01':
+            site_map_station = ['CBG', 'SAG', 'ROT', 'COF']
+
+            if any(s == platform_code for s in site_map_station):
+                acorn_file_type = "gridded_1h-avg-wave-site-map_QC"
+            else:
+                acorn_file_type = "gridded_1h-avg-wave-station-map_QC"
+
+        elif product_type == '1-hour-avg' and file_version == 'FV00':
+            acorn_file_type = "gridded_1h-avg-current-map_non-QC"
+
+        elif product_type == '1-hour-avg' and file_version == 'FV01':
+            acorn_file_type = "gridded_1h-avg-current-map_QC"
+
+        else:
+            unknown_product = True
+    else:
+        unknown_product = True
+
+    if unknown_product:
+        raise InvalidFileNameError("file name: \"{filename}\" Unknown product type from filename".
+                                   format(filename=os.path.basename(filepath)))
+
+    return acorn_file_type
+
+
+class AcornHandler(HandlerBase):
+    def __init__(self, *args, **kwargs):
+        super(AcornHandler, self).__init__(*args, **kwargs)
+        self.allowed_extensions = ['.nc']
+        self.opendap_root = self.config.pipeline_config['global'].get('opendap_root')
+
+    def preprocess(self):
+        nc_collection = self.file_collection[0]
+
+        # enforce a check on the include_regexes and the filename to make sure the handler is processing the correct
+        # matching rexep file written in the config file
+        m = re.search(self.include_regexes[0], nc_collection.name)
+        if m is None:
+            raise InvalidFileNameError(
+                "file name: \"{filename}\" not allowed in this pipeline".format(
+                    filename=os.path.basename(nc_collection.name)))
+
+        # if file contains any of FILE_TYPE_NEED_INDEX, we index, otherwise publish only
+        if any(s in get_type(nc_collection.name) for s in FILE_TYPE_NEED_INDEX):
+            nc_collection.publish_type = PipelineFilePublishType.HARVEST_UPLOAD
+        else:
+            nc_collection.publish_type = PipelineFilePublishType.UPLOAD_ONLY
+
+        # check if file with same dest path already on s3. If yes, check its date_created nc attribute to know
+        # if we need to overwrite this object or not
+        destination_s3 = self.dest_path(nc_collection.name)
+
+        storage_query_res = self.state_query.query_storage(destination_s3)
+        if storage_query_res:
+            # creation date of the new file in the pipeline
+            creation_date_nc_pipeline = self.get_creation_date(nc_collection.src_path)
+
+            # creation date of the file already published
+            creation_date_nc_s3 = self.get_creation_date(os.path.join(self.opendap_root, destination_s3))
+
+            if creation_date_nc_pipeline < creation_date_nc_s3:
+                nc_collection.publish_type = PipelineFilePublishType.NO_ACTION
+
+    @staticmethod
+    def get_creation_date(netcdf_path):
```

Comment:
This should be moved out of the class into a separate function, since it is static and not related directly to the class.